x-pack/filebeat/input/httpjson/config_auth.go

```diff
@@ -109,14 +111,30 @@ func (o *oAuth2Config) client(ctx context.Context, client *http.Client) (*http.C
 
 	switch o.getProvider() {
 	case oAuth2ProviderAzure, oAuth2ProviderDefault:
-		creds := clientcredentials.Config{
-			ClientID:       o.ClientID,
-			ClientSecret:   o.ClientSecret,
-			TokenURL:       o.getTokenURL(),
-			Scopes:         o.Scopes,
-			EndpointParams: o.getEndpointParams(),
+		if o.User != "" || o.Password != "" {
```

Comment:
Is it also valid for Azure?

Do you think it's worth considering introducing another type: `oauth2ProviderSalesforce`?

Reply:
I am not sure but they do support username-password authentication [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth-ropc#authorization-request). 
If its valid then that means they will also be able to use another authentication mechanism.
I will look into it if its not ok then we can add oauth2ProviderSalesforce

Reply:
Do you think it's worth focusing on the oauth2ProviderSalesforce provider now?

Reply:
I think we have another option.
what if we use oAuth2ProviderDefault:
```
	case oAuth2ProviderDefault:
		if o.User != "" || o.Password != "" {
			conf := &oauth2.Config{
				ClientID:     o.ClientID,
				ClientSecret: o.ClientSecret,
				Endpoint: oauth2.Endpoint{
					TokenURL:  o.TokenURL,
					AuthStyle: AuthStyleInParams,
				},
			}
			token, err := conf.PasswordCredentialsToken(ctx, o.User, o.Password)
			if err != nil {
				return nil, fmt.Errorf("oauth2 client: error loading credentials using user and password: %w", err)
			}
			return conf.Client(ctx, token), nil
		} else {
			creds := clientcredentials.Config{
				ClientID:       o.ClientID,
				ClientSecret:   o.ClientSecret,
				TokenURL:       o.getTokenURL(),
				Scopes:         o.Scopes,
				EndpointParams: o.getEndpointParams(),
			}
			return creds.Client(ctx), nil
		}
	case oAuth2ProviderAzure:
		creds := clientcredentials.Config{
			ClientID:       o.ClientID,
			ClientSecret:   o.ClientSecret,
			TokenURL:       o.getTokenURL(),
			Scopes:         o.Scopes,
			EndpointParams: o.getEndpointParams(),
		}
		return creds.Client(ctx), nil
```

What do you think if we do this?
Instead creating new provider we can use default one and user will also able use user-passwrod method other than salesforce.

Reply:
RFC defines it (Resource Owner Password Credentials Grant), so I don't see a reason why don't apply it also here. Feel free to modify the code.